src/lib/qnio/nio_server.c

```diff
@@ -452,14 +452,16 @@ qns_server_start(char *node, char *port)
                             {
                                 nioDbg("SSL_error is want read or want write %d", ret);
                                 /* put fd in epoll loop to try again */
-                                event.events = EPOLLIN;
+                                event.events = EPOLLIN | EPOLLOUT;
```

Comment:
EPOLLOUT should be rare.  It occurs when the socket sndbuf is exhausted - then the kernel network stack has to tell the application to hold off until space becomes available again.